igor-web/src/main/groovy/com/netflix/spinnaker/igor/gitlabci/service/GitlabCiService.java

```diff
@@ -175,17 +176,20 @@ private Map<String, Object> getPropertyFileFromLog(String projectId, Integer pip
 
             return properties;
 
-          } catch (RetrofitError e) {
-            // retry on network issue, 404 and 5XX
-            if (e.getKind() == RetrofitError.Kind.NETWORK
-                || (e.getKind() == RetrofitError.Kind.HTTP
-                    && (e.getResponse().getStatus() == 404
-                        || e.getResponse().getStatus() >= 500))) {
+          } catch (SpinnakerNetworkException e) {
+            // retry on network issue
+            throw e;
+          } catch (SpinnakerHttpException e) {
+            // retry on 404 and 5XX
+            if (e.getResponseCode() == 404 || e.getResponseCode() >= 500) {
               throw e;
             }
-            SpinnakerException ex = new SpinnakerException(e);
-            ex.setRetryable(false);
-            throw ex;
+            e.setRetryable(false);
+            throw e;
+          } catch (SpinnakerServerException e) {
```

Comment:
I suppose this is faithful to the original behavior, but it doesn't seem right...a note for our future selves.